collections/indexes/multi.go

```diff
@@ -121,6 +123,40 @@ func (m *Multi[ReferenceKey, PrimaryKey, Value]) MatchExact(ctx context.Context,
 	return m.Iterate(ctx, collections.NewPrefixedPairRange[ReferenceKey, PrimaryKey](refKey))
 }
 
+// RefKeys returns a list of all the MultiIterator's reference keys (may contain duplicates).
+// Enable the "unique" argument to get a unique list of reference keys (the reference key must be comparable)
+func (m *Multi[ReferenceKey, PrimaryKey, Value]) RefKeys(ctx context.Context, unique bool) ([]ReferenceKey, error) {
+	// sanity check - enabled unique with non-comparable ReferenceKey type
+	if unique && !reflect.ValueOf((*ReferenceKey)(nil)).Comparable() {
+		return nil, fmt.Errorf("cannot retrieve unique reference keys since type is not comparable: %T", reflect.TypeOf((*ReferenceKey)(nil)))
+	}
+
+	iter, err := m.refKeys.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := []ReferenceKey{}
+	visited := map[interface{}]struct{}{}
```

Comment:
i do not think this will work for all key kinds

Reply:
it'll work because I check if the key is comaprable